estate/models/estate_property.py

```diff
@@ -0,0 +1,66 @@
+from odoo import fields, models, api
+
+from datetime import datetime, date
+from dateutil.relativedelta import relativedelta
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property App"
+
+    name= fields.Char("Estate property name", required=True)
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Available From", copy=False, default=fields.Datetime.now() + relativedelta(months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area")
+    garden_orientation = fields.Selection(
+        string="Orientation",
+        selection=[("north", "North"), ("south", "South"), ("east", "East"), ("west", "West")],
+        help="Orientation is used to define the orientation")
+
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        string="States",
+        selection=[("new", "New"), ("offer received", "Offer Received"), ("offer accepted", "Offer Accepted"), ("sold", "Sold"), ("canceled", "Canceled")],
+        default="new",
+        help="Choose the state of the Offer")
+
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    salesperson_id = fields.Many2one('res.users', string='Salesman', default=lambda self: self.env.user)
+
+    tag_ids = fields.Many2many("estate.property.tag")
+
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+
+    total_area = fields.Float(compute="_compute_total_area")
+
+    best_price = fields.Float(compute="_calculate_best_price", string="Best Offer")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids")
+    def _calculate_best_price(self):
+        self.best_price = 0
+        for record in self.offer_ids:
+            if self.best_price < record.price:
+               self.best_price = record.price
+
```

Comment:
it works as expected; possible amelioration would be to use max() and mapped() functions => get the price of all the offers and pick the top one. Bit more `pythonesque`